packages/react/src/suppress-safari-class-name-mismatch.js

```diff
@@ -0,0 +1,105 @@
+function classNameBelongsToCache(className, cacheKey) {
+  return className.startsWith(cacheKey + '-')
+}
+
+function getLabelFromClassName(className, cacheKey) {
+  const withoutCacheKey = className.replace(cacheKey + '-', '')
+  const parts = withoutCacheKey.split('-')
+
+  return parts.length === 2 ? parts[1] : undefined
+}
+
+export function isPossiblyBenignClassNameMismatch(
+  cache,
+  serverClassName,
+  browserClassName
+) {
+  const serverLabel = getLabelFromClassName(serverClassName, cache.key)
+  const browserLabel = getLabelFromClassName(browserClassName, cache.key)
+
+  const browserIsMissingLabel =
+    typeof serverLabel === 'string' && typeof browserLabel === 'undefined'
+
+  const browserHasDifferentLabel =
+    typeof serverLabel === 'string' &&
+    typeof browserLabel === 'string' &&
+    serverLabel !== browserLabel
+
+  return (
+    classNameBelongsToCache(serverClassName, cache.key) &&
+    classNameBelongsToCache(browserClassName, cache.key) &&
+    (browserIsMissingLabel || browserHasDifferentLabel)
+  )
+}
+
+function getStackTrace() {
+  return new Error().stack
+}
+
+function __properTailCallTestFunction() {
+  return getStackTrace()
+}
+
+const jsEngineDoesProperTailCalls = !__properTailCallTestFunction().includes(
```

Comment:
i would wrap the whole thing into an IIFE like this:
```suggestion
const jsEngineDoesProperTailCalls = /* #__PURE__ */ (() => { /* move the whole logic required for this check here */ })()
```

it will make sure that this gets dropped in prod